test.js

```diff
@@ -29,7 +35,7 @@ test('returns `undefined` when nothing could be found', async t => {
 	t.is((await m([1, 2, 3], () => false)), undefined);
 });
 
-test('rejected return value in `tester` rejects the promise', async t => {
-	const fixtureErr = new Error('fixture');
-	await t.throws(m([1, 2, 3], () => Promise.reject(fixtureErr)), fixtureErr.message);
-});
+// test('rejected return value in `tester` rejects the promise', async t => {
+// 	const fixtureErr = new Error('fixture');
+// 	await t.throws(m([1, 2, 3], () => Promise.reject(fixtureErr)), fixtureErr.message);
+// });
```

Comment:
Just writing down what I see.

It's definitely caused by [this](https://github.com/sindresorhus/p-locate/blob/preserve-order/index.js#L24) line, more specifically:

```js
limit(() => Promise.resolve(el).then(tester))
```

If you write the `finder` function to something like this

```js

const finder = el => Promise.all(el)
    .then(val => val[1] === true && Promise.reject(new EndError(val[0])))
    .catch(() => console.log('catched'));
```

It doesn't fail with an unhandled rejection. This indeed is very odd because the finder is wrapped inside a promise chain which on his turn is catched.